### PR TITLE
Online DDL: unsupporting `gh-ost` DDL strategy

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

See https://github.com/vitessio/vitess/issues/15692

This PR:

- Removes the embedded `gh-ost` binary from `vttablet` (the user will have to supply their own `gh-ost` binary, and unless it's `/usr/bin/gh-ost` they will have to run `vttablet --gh-ost-path`)
- Removes `onlineddl_ghost` endtoend tests, and replaces `gh-ost` strategy tests with `vitess` strategy in any other tests.

With that, we turn `gh-ost` to be _unsupported_. Otherwise `gh-ost` strategy still works.

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/15692
- Documentation: https://github.com/vitessio/website/pull/1725

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Docs PR: https://github.com/vitessio/website/pull/1725

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -89,8 +89,8 @@ var acceptableDropTableIfExistsErrorCodes = []sqlerror.ErrorCode{sqlerror.ERCant
 var copyAlgorithm = sqlparser.AlgorithmValue(sqlparser.CopyStr)
 
 var (
-	ghostOverridePath       string
-	ptOSCOverridePath       string
+	ghostBinaryPath         = "/usr/bin/gh-ost"
+	ptOSCBinaryPath         = "/usr/bin/pt-online-schema-change"
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:93]: `pt-online-schema-change` remains on `/usr/bin` as this was the previous behavior. Not changing it, will just remove it when the time comes.

**Review by dbussink — APPROVED** [review]

**Review by mattlord — APPROVED** [review]
LGTM! We need to update a couple of test output files now. I'll push a quick commit for that to save you the CI time when you're ready to merge.

**Comment by shlomi-noach** [comment]
@mattlord thank you!

**Pull request merged**